XML_to_SON/grammar_parser_test.py

```diff
@@ -137,28 +208,32 @@ def handle_choice_element(node):
 def process_schema_from_mjson(xml_string, element_name):
     root = ET.fromstring(xml_string)
     processed_content = process_node(root)
-    
     return {element_name: processed_content}
 
 def integrate_detailed_schemas(final_json, processed_facilities, processed_regions, processed_institutions):
-    for facility_name, facility_config in processed_facilities.items():
-        if "facility" in final_json["simulation"] and facility_name in final_json["simulation"]["facility"]["config"]["ChildExactlyOne"]:
-            new_facility_config = {"InputTmpl": '"'+facility_name+'"'}
-            new_facility_config.update(facility_config)
-            final_json["simulation"]["facility"]["config"][facility_name] = new_facility_config
-
-    for region_name, region_config in processed_regions.items():
-        if "region" in final_json["simulation"] and region_name in final_json["simulation"]["region"]["config"]["ChildExactlyOne"]:
-            new_region_config = {"InputTmpl": '"' + region_name + '"'}
-            new_region_config.update(region_config)
-            final_json["simulation"]["region"]["config"][region_name] = new_region_config
-
+    for agent_type, agent_list in zip(['facility', 'region'], [processed_facilities, processed_regions]):
```

Comment:
It might be safer/more readable to manually bind/zip these since there will only ever be two:
```suggestion
    for agent_type, agent_list in [ ('facility, processed_facilities), ('region', processed_regions) ]:
```